tests/sbc_test.py

```diff
@@ -76,6 +76,80 @@ def simulator(theta):
     get_nltp(thetas, xs, posterior)
 
 
+@pytest.mark.slow
+@pytest.mark.parametrize("prior", ("boxuniform", "independent"))
+@pytest.mark.parametrize("method, sampler", [(SNLE, "vi")])
+def test_consistent_sbc_results(method, prior, sampler, model="mdn"):
+    """Tests running inference and then SBC and obtaining nltp."""
+
+    num_dim = 2
+    if prior == "boxuniform":
+        prior = BoxUniform(-torch.ones(num_dim), torch.ones(num_dim))
+    else:
+        prior = MultipleIndependent([
+            Uniform(-torch.ones(1), torch.ones(1)) for _ in range(num_dim)
+        ])
+
+    num_simulations = 200
+    max_num_epochs = 20
+    num_sbc_runs = 4
+
+    likelihood_shift = -1.0 * ones(num_dim)
+    likelihood_cov = 0.3 * eye(num_dim)
+
+    def simulator(theta):
+        return linear_gaussian(theta, likelihood_shift, likelihood_cov)
+
+    inferer = method(prior, show_progress_bars=False, density_estimator=model)
+
+    theta, x = simulate_for_sbi(simulator, prior, num_simulations)
+
+    _ = inferer.append_simulations(theta, x).train(
+        training_batch_size=100, max_num_epochs=max_num_epochs
+    )
+    if method == SNLE:
+        posterior_kwargs = {
+            "sample_with": "mcmc" if sampler == "mcmc" else "vi",
+            "mcmc_method": "slice_np_vectorized",
+            "mcmc_parameters": {"num_chains": 10, "thin": 5, "warmup_steps": 10},
+        }
+    else:
+        posterior_kwargs = {}
+
+    posterior = inferer.build_posterior(**posterior_kwargs)
+    num_posterior_samples = 250
```

Comment:
```suggestion
    num_posterior_samples = 1000
```
we need enough samples to be consistent, and sampling is cheap for SNPE.

Reply:
see above about my point of NPE vs SNLE.

Reply:
addressed with 89e1ffd1